app/src/main/java/com/example/tasky/feature_authentication/data/remote/TaskyAuthApi.kt

```diff
@@ -0,0 +1,31 @@
+package com.example.tasky.feature_authentication.data.remote
+
+import com.example.tasky.feature_authentication.data.remote.auth.LoginRequest
+import com.example.tasky.feature_authentication.data.remote.auth.LoginResponse
+import com.example.tasky.feature_authentication.data.remote.auth.RegistrationRequest
+import retrofit2.Response
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.Header
+import retrofit2.http.POST
+
+interface TaskyAuthApi {
+    @POST("/register")
+    suspend fun signUp(
+        @Header("x-api-key") key: String,
```

Comment:
The API key will be needed for every single request. In that case it makes sense to add an interceptor. That's something that lets you manipulate every request before it's fired (for example to add a header like here). This is how it can be done:
```
class ApiKeyInterceptor(
    private val apiKey: String
): Interceptor {

    override fun intercept(chain: Interceptor.Chain): Response {
        return chain.proceed(
            chain
                .request()
                .newBuilder()
                .addHeader("x-api-key", apiKey)
                .build()
        )
    }
}
```
And then add it when creating the API instance:
```
@Provides
@Singleton
fun provideOkHttpClient(
    apiKeyInterceptor: ApiKeyInterceptor,
): OkHttpClient {
    return OkHttpClient.Builder()
        .addInterceptor(ApiKeyInterceptor(BuildConfig.API_KEY))
        .build()
}

@Provides
@Singleton
fun provideAuthApi(client: OkHttpClient): AuthApi {
    return Retrofit.Builder()
        .baseUrl(BASE_URL)
        .addConverterFactory(MoshiConverterFactory.create())
        .client(client) // <-- Here, you add the client with the interceptor
        .build()
        .create()
}
```